utils/R/convert.input.R

```diff
@@ -138,57 +217,100 @@ convert.input <- function(input.id, outfolder, formatname, mimetype, site.id, st
-       } 
-    } else {
-       args = c(dbfile$file_path,dbfile$file_name,outfolder,start_date,end_date) 
-       cmdFcn  = paste0(pkg,"::",fcn,"(",paste0("'",args,"'",collapse=","),",format=",paste0(list(format.vars)),")")
-    } 
-    print(cmdFcn) #do we want to print this?
+  } else if (conversion == "local.remote") { 
+    # perform conversion on local or remote host
+
+    fcn.args <- input.args
+    fcn.args$overwrite <- overwrite
+    fcn.args$in.path <- dbfile$file_path
+    fcn.args$in.prefix <- dbfile$file_name
+    fcn.args$outfolder <- outfolder
+    fcn.args$start_date <- start_date
+    fcn.args$end_date <- end_date 
+
+    arg.string <- listToArgString(fcn.args)
+    
+    if(!missing(format.vars)) {
+      arg.string <- paste0(arg.string, ", format=", paste0(list(format.vars)))
+    }
+    
+    cmdFcn = paste0(pkg, "::", fcn, "(", arg.string, ")")
+    logger.debug(paste0("convert.input executing the following function:\n", cmdFcn))
 
     result <- remote.execute.R(script=cmdFcn, host, user=NA, verbose=TRUE, R="R")
   }
   
   print("RESULTS: Convert.Input")
   print(result)
   print(names(result))
-  
+
   #--------------------------------------------------------------------------------------------------#
   # Insert into Database
-  
+
   # Use existing site, unless otherwise specified (ex: subsetting case, using newsite)
-  if("newsite" %in% names(l) && is.null(l[["newsite"]])==FALSE){
-    siteid <- l$newsite
-  }else{
+  if("newsite" %in% names(input.args) && !is.null(input.args[["newsite"]])){
+    siteid <- input.args$newsite
+  } else {
     siteid <- site.id
   }
   
   outlist <- unlist(strsplit(outname,"_")) 
-  
+
   ## insert new record into database
-  if(write==TRUE){
+  if(write==TRUE) {
+    if(exists("existing.input") && nrow(existing.input) > 0 &&
+       (existing.input$start_date != start_date || existing.input$end_date != end_date)) {
+      db.query(paste0(
+        "UPDATE inputs SET start_date='", start_date, "', end_date='", end_date, "', ", 
+        "updated_at=NOW() WHERE id=", existing.input$id), con)
+    }
+
+    parent.id <- ifelse(is.null(input), NA, input$id)
     newinput <- dbfile.input.insert(in.path=dirname(result$file[1]),
                                     in.prefix=result$dbfile.name[1],
                                     siteid = siteid, 
-                                    startdate = startdate, 
-                                    enddate = enddate, 
+                                    startdate = start_date, 
+                                    enddate = end_date, 
                                     mimetype, 
                                     formatname,
-                                    parentid = input$id,
+                                    parentid = parent.id,
                                     con = con,
                                     hostname = machine$hostname) 
+
+    successful <- TRUE
```

Comment:
See above for the two alternative functions that will be run by `on.exit` depending on whether we make it all the way here and successful==TRUE.